extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRecorder.java

```diff
@@ -58,10 +58,16 @@ public void doStartActions() {
                             liquibase.dropAll();
                         }
                         if (config.migrateAtStart) {
-                            if (config.validateOnMigrate) {
-                                liquibase.validate();
+                            var lockService = LockServiceFactory.getInstance().getLockService(liquibase.getDatabase());
+                            try {
+                                lockService.waitForLock();
+                                if (config.validateOnMigrate) {
```

Comment:
So we're adding the overhead(?) of locking even in the case where we don't validate.

But given `validate-on-migrate` ist true by default it should be alright.

Reply:
The overhead should be minimal, as the first thing `liquibase.update()` does is to acquire the lock. But if it already has it, it will result more or less in a map lookup, this should not generate much overhead.